src/python/zquantum/core/measurement.py

```diff
@@ -500,7 +514,42 @@ def get_expectation_values(
                     value = -float(count) / num_measurements
                 expectation += np.real(coefficient) * value
             expectation_values.append(np.real(expectation))
-        return ExpectationValues(np.array(expectation_values))
+
+        correlations = np.zeros((len(ising_operator.terms),) * 2)
```

Comment:
It seems to me that you've added a lot of logic into this function and it's a little bit bloated – perhaps consider decomposing it into several functions to make it more legible?

Reply:
I've moved some duplicated logic into a new function `get_expectation_value_from_frequencies` and implemented @aakunitsa's suggestion to simplify the calculation of covariances.